New features: override version, detect git URLs, ... (#29517)
LLNL/spack: lib/spack/docs/packaging_guide.rst

   $ spack create https://gmplib.org/download/gmp/gmp-6.1.2.tar.bz2

Spack examines the tarball URL and tries to figure out the name and version
of the package to be created. If the name contains uppercase letters, these
are automatically converted to lowercase. If the name contains underscores
or periods, these are automatically converted to dashes.

.. note::

>     You can override version detected by Spack with `--version` argument

Reviewer: ```suggestion
    You can override the version detected by Spack using the `--version` argument.
```

     version('1.0.1', tag='v1.0.1', submodules=True)

.. note::

>     Spack will detect git-like url when you run `spack create` by checking

Reviewer: ```suggestion
    Spack will detect a git-like url when you run `spack create` by checking
```

>     for `git@` prefix or `.git` suffix. One can also use `--git` flag to signal

Reviewer: ```suggestion
    for `git@` prefix or `.git` suffix. Or you can use `--git` flag to signal
```

    Spack that the URL passed to `spack create` is not a direct URL, but a Git
    repository URL.
    

>     You can also specify tag (with `--tag` argument), commit (`--commit`) or branch (`--branch`). By default,

Reviewer: ```suggestion
    You can also specify a tag (with `--tag` argument), a commit (`--commit`) or a branch (`--branch`). By default,
```

>     Spack will use branch or tag name, or commit id as version, you can override it with `--version` argument.

Reviewer: ```suggestion
    Spack will use the branch, tag name, or commit id as the version, which you can override with the `--version` argument.
```
